src/pages/Customer/Cart/CartDesign/CartPage.js

```diff
@@ -1,21 +1,72 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+
+import { getDetailsOfCart } from "api/user/cart";
 
 import styles from "./CartPage.module.scss";
 
-import CartData from "../CartData";
 import CartMainCard from "../CartMainCard";
+import Spinner from "components/Spinner/Spinner";
 
 const CartPage = () => {
+  const [cartData, setCartData] = useState([]);
+  const [isCartDataFetched, setIsCartDataFetched] = useState(false);
+  const [shippingCharge, setShippingCharge] = useState(0);
+  let total = 0;
+
+  const calcFirstBill = () => {
+    if (cartData.length === 0) return 0;
+    const priceArr = cartData.map(
+      (item) => item.quantity * item.refProduct.availabilities[0].price
+    );
+    total = priceArr.reduce((acc, curr) => acc + curr);
+    return total;
+  };
+
+  const updateFinalBill = (productId, quantity) => {
+    const dummyCartData = [...cartData];
+    const index = dummyCartData.findIndex((item) => item._id === productId);
+    if (index < 0) return;
+    dummyCartData[index].quantity = quantity;
+    if (quantity === -1) {
+      dummyCartData.splice(index, 1);
+    }
+    setCartData(dummyCartData);
+  };
+
+  useEffect(() => {
+    getDetailsOfCart().then((res) => {
+      if (!res) {
+        setIsCartDataFetched(false);
+        setShippingCharge(0);
+      }
+      if (res.data?.products.length) setShippingCharge(50);
+      setCartData(res.data?.products);
+      setIsCartDataFetched(true);
+    });
+  }, []);
+
+  const totalPrice = calcFirstBill();
   return (
     <div className={styles.outer}>
       <div className={styles.leftPortion}>
         <div className={styles.shopTitle}>
-          Your Shopping Cart <span>(2 Items)</span>
+          Your Shopping Cart <span>({cartData.length} Items)</span>
         </div>
         <div className={styles.leftInner}>
-          {CartData.map((Items) => {
-            return <CartMainCard key={Items.productId} items={Items} />;
-          })}
+          {!isCartDataFetched ? (
+            <Spinner />
+          ) : (
+            cartData.length !== 0 &&
+            cartData.map((item) => {
+              return (
+                <CartMainCard
+                  key={item._id}
+                  item={item}
+                  refreshCartData={updateFinalBill}
+                />
+              );
+            })
+          )}
```

Comment:
if user dont have anything in cart then we should show user something message with some svg(undraw.co)